src/cocoa_joystick.m

```diff
@@ -496,9 +508,9 @@ int _glfwPlatformJoystickPresent(int joy)
 const char* _glfwPlatformGetJoystickName(int joy)
 {
     _GLFWjoydeviceNS* joystick = _glfw.ns_js.devices + joy;
-    if (!pollJoystickEvents(joystick))
+    if (joystick->present)
+        return joystick->name;
+    else
```

Comment:
When a test case is most likely true as it is here, I like to handle the successful case first because it's generally more efficient: (1) it avoids the cost of the branch and (2) it avoids the cost of loading the instruction cache when large instruction blocks follow the condition.  That's a fairly ingrained habit, and I failed to notice that the code followed the pattern of handling failure cases first.